src/test/java/io/confluent/connect/jdbc/source/JdbcSourceTaskUpdateTest.java

```diff
@@ -863,6 +866,81 @@ public void testCustomQueryWithTimestamp() throws Exception {
     PowerMock.verifyAll();
   }
 
+  @Test
+  public void testCustomQueryMultipleRecordsWithDBPartition() throws Exception {
```

Comment:
@ManasjyotiSharma - I see that this test doesn't actually cover the case we discuss in the PR description. I'll rework it to make sure it's the same/similar.